src/transactions/feed/TransactionFeedV2.tsx

```diff
@@ -95,18 +96,32 @@ function sortTransactions(transactions: TokenTransaction[]): TokenTransaction[]
  * Otherwise, if we merge all the stand by transactins into the page it will cause more late transactions
  * that were already merged to be removed from the top of the list and move them to the bottom.
  * This will cause the screen to "shift", which we're trying to avoid.
+ *
+ * Note: when merging the first page – stand by transactions might include some new pending transaction.
+ * In order to include them in the merged list we need to also check if the stand by transaction is newer
+ * than the max timestamp from the page. But this must only happen for the first page as otherwise any
+ * following page would include stand by transactions from previous pages.
  */
-function mergeStandByTransactionsInRange(
-  transactions: TokenTransaction[],
-  standBy: TokenTransaction[]
-): TokenTransaction[] {
+function mergeStandByTransactionsInRange({
+  transactions,
+  standByTransactions,
+  currentCursor,
+}: {
+  transactions: TokenTransaction[]
+  standByTransactions: TokenTransaction[]
+  currentCursor?: number
+}): TokenTransaction[] {
   if (transactions.length === 0) return []
 
   const allowedNetworks = getAllowedNetworksForTransfers()
   const max = transactions[0].timestamp
   const min = transactions.at(-1)!.timestamp
 
-  const standByInRange = standBy.filter((tx) => tx.timestamp >= min && tx.timestamp <= max)
+  const standByInRange = standByTransactions.filter((tx) => {
+    const inRange = tx.timestamp >= min && tx.timestamp <= max
+    const newTransaction = currentCursor === FIRST_PAGE_TIMESTAMP && tx.timestamp > max
```

Comment:
With the change to how the cursor is structured  in blockchain-api (using the b64-encoded Zerion next page URL), it seems like the cursor semantics in the wallet need to change accordingly, right?